solrsearch-plugins/src/main/java/co.cask.hydrator.plugin/common/SolrSearchSinkUtility.java

```diff
@@ -0,0 +1,116 @@
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package co.cask.hydrator.plugin.common;
+
+import co.cask.cdap.api.data.schema.Schema;
+import org.apache.solr.client.solrj.SolrClient;
+import org.apache.solr.client.solrj.impl.CloudSolrClient;
+import org.apache.solr.client.solrj.impl.HttpSolrClient;
+
+/**
+ * Utility class for SolrSearchSink and RealtimeSolrSearchSink plugins.
+ */
+public class SolrSearchSinkUtility {
+
+  private static final String SINGLE_NODE_SOLR_MODE = "Single Node Solr";
+  private static final String SOLR_CLOUD_MODE = "SolrCloud";
+
+  /**
+   * Returns the SolrClient for establishing the connection with Solr server, using the hostname and port provided by
+   * user.
+   *
+   * @return Solr client
+   */
+  public static SolrClient getSolrConnection(SolrSearchSinkConfig config) {
+    String urlString;
+    SolrClient solrClient = null;
+    if (config.getSolrMode().equals(SINGLE_NODE_SOLR_MODE)) {
+      urlString = "http://" + config.getSolrHost() + "/solr/" + config.getCollectionName();
+      solrClient = new HttpSolrClient(urlString);
+    } else if (config.getSolrMode().equals(SOLR_CLOUD_MODE)) {
+      CloudSolrClient solrCloudClient = new CloudSolrClient(config.getSolrHost());
+      solrCloudClient.setDefaultCollection(config.getCollectionName());
+      solrClient = solrCloudClient;
+    }
+    return solrClient;
+  }
+
+  /**
+   * Validates whether the host entered for Single Node Solr instance is proper or not.
+   */
+  public static void validateSolrConnectionString(SolrSearchSinkConfig config) {
+    if (config.getSolrMode().equals(SINGLE_NODE_SOLR_MODE) && config.getSolrHost().contains(",")) {
+      throw new IllegalArgumentException(String.format("Multiple hosts '%s' found for Single Node Solr.",
+                                                       config.getSolrHost()));
+    }
+  }
+
+  /**
+   * Verifies whether the Solr server is running or not. Also, validates whether the collection provided by user is
+   * registered with the Solr server or not.
+   */
+  public static void verifySolrConfiguration(SolrSearchSinkConfig config) {
+    SolrClient client = getSolrConnection(config);
+    try {
+      client.ping();
+    } catch (Exception e) {
+      throw new IllegalArgumentException(
+        String.format("Server refused connection at '%s'. Please make sure that either the Solr/Zookeeper services " +
+                        "are properly running or the collection '%s' exists in the Solr Server.", config.getSolrHost(),
+                      config.getCollectionName()));
+    }
+  }
+
+  /**
+   * Validates whether the Id field provided by user is present in the input schema or not.
+   *
+   * @param inputSchema
+   */
+  public static void validateIdField(Schema inputSchema, SolrSearchSinkConfig config) {
+    if (inputSchema != null && inputSchema.getField(config.getIdField()) == null) {
+      throw new IllegalArgumentException(
```

Comment:
also need to check that the id field is not nullable.  I would imagine there are also type restrictions on what can be an id right?